### ISSUE TITLE
return-undefined false positive with async function

### ISSUE THREAD

**Issue by nicolashenry** [palantir/tslint#3234]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.2
- __Running TSLint via__: CLI / VSCode

#### TypeScript code being linted

```ts
declare function test(cb: () => Promise<void> | void): void;

test(async () => {
	return;
});
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "warning",
  "rules": {
    "return-undefined": true
  }
}
```

#### Actual behavior

```
[tslint] Value-returning function should use `return undefined;`, not just `return;`. (return-undefined)
```

#### Expected behavior

No error
